.github/workflows/gradle.yml

```diff
@@ -27,19 +27,20 @@ jobs:
         run: ./run-checks.sh
 
       - name: Validate Gradle Wrapper
-        uses: gradle/wrapper-validation-action@v1
+        uses: gradle/actions/wrapper-validation@v3
 
-      - name: Setup JDK 11
-        uses: actions/setup-java@v1
+      - name: Setup JDK 17
+        uses: actions/setup-java@v4
         with:
-          java-version: '11'
+          distribution: 'zulu'
+          java-version: '17'
           java-package: jdk+fx
 
       - name: Build and check with Gradle
         run: ./gradlew check coverage
 
       - name: Upload coverage reports to Codecov
         if: runner.os == 'Linux'
-        uses: codecov/codecov-action@v3
-        env:
-          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        uses: codecov/codecov-action@v4.0.1
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
@brein62 Some of these changes are not strictly needed for the Java 17 migration. If you think they are still useful, you can send separate PRs for them, with the justification.